core/xchaincore.go

```diff
@@ -113,6 +116,10 @@ type XChainCore struct {
 	txidCacheExpiredTime time.Duration
 	enableCompress       bool
 	pruneOption          config.PruneOption
+
+	// cache for duplicate block message
+	msgCache           *common.LRUCache
```

Comment:
common.LRUCache is more expensive than go-cache.

Reply:
Here I want to limit cache item by capacity not expire time. Using capacity could keep cache in a reasonable memory size, and default 1000 blocks is enough for most case. Since new block is a low frequency message, so the CPU cost could be ignored.